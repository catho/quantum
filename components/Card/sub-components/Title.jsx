import PropTypes from 'prop-types';
import styled from 'styled-components';

import { baseFontSize as defaultBaseFontSize } from '../../shared/theme';

const Heading = styled.h2`
  font-size: ${({ small, theme: { baseFontSize } }) =>
    `${small ? baseFontSize * 1.25 : baseFontSize * 1.5}`}px;
  font-weight: 600;
  margin: 0;
`;

const Title = ({
  small = false,
  theme = {
    baseFontSize: defaultBaseFontSize,
  },
  ...props
}) => <Heading small={small} theme={theme} {...props} />;

Title.displayName = 'Card.Title';

Title.propTypes = {
  /** default `font-size` is `24px`, with `small` prop defined the `font-size` is changed to `20px`. */
  small: PropTypes.bool,
  theme: PropTypes.shape({
    baseFontSize: PropTypes.number,
  }),
};

export default Title;
