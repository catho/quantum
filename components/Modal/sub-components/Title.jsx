import classNames from 'classnames';
import Card from '../../Card';
import styles from './Title.module.css';

const Title = ({ className = '', children, ...props }) => {
  const classContent = classNames(styles.title, className);
  return (
    <Card.Title className={classContent} {...props}>
      {children}
    </Card.Title>
  );
};

export default Title;
