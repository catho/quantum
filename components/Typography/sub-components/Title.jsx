import classNames from 'classnames';

const Title = ({ className = '', children, ...props }) => {
  const classContent = classNames(className);
  return (
    <h1 className={classContent} {...props}>
      {children}
    </h1>
  );
};

export default Title;
