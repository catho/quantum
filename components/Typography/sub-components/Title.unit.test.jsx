import { render } from '@testing-library/react';
import Title from './Title';

describe('<Title />', () => {
  it('Should match the snapshot', () => {
    expect(render(<Title />).asFragment()).toMatchSnapshot();
  });
});
