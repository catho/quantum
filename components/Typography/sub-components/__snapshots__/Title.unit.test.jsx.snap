// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> Should match the snapshot 1`] = `
<DocumentFragment>
  <h1
    class=""
  />
</DocumentFragment>
`;
